### DISCUSSION TITLE
What's the idiomatic way of assigning plugin subsets to workspaces?

### DISCUSSION THREAD

**Discussion by pkovacs** [bufbuild/buf#3089] [Q&A]
If I am using 2 plugins, one for generating cpp protobuf stubs and the other for golang, and I partition my protos into two distinct, non-overlapping workspaces, call them A and B -- what is the idiomatic way of assigning the cpp plugin to workspace A and the golang plugin to workspace B?   Can this be done in one buf context or must I create two contexts, each with its own buf.gen.yaml?

**Comment by emcfarlane** [comment]
Specifying two `buf.gen.yaml` with different inputs targeting the A/B module or paths is the recommended workflow. See docs for [`buf.gen.yaml`](https://buf.build/docs/configuration/v2/buf-gen-yaml#inputs).
